pkg/apis/extensions/types.go

```diff
@@ -226,6 +226,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
```

Comment:
Discussed offline. 

The reason this is here is because the actual rollback of the pod template is executed in the Deployment controller, not in the registry. That division of responsibility was chosen due to the complexity and cost of this operation. The controller already reads all the RCs and can cache which RCs correspond to which Deployment.

This is sort of an internal API between the apiserver and Deployment controller, so we discussed whether it should be conveyed via an annotation instead, but is also useful for the client, as a way to determine whether the rollback has been acted upon asynchronously yet or not. We will also want to generate an event once the rollback executes, conveying success or failure. Status alone could be quickly overwritten by another mutation.

As for single field vs nested struct: We don't expect users or even clients to specify this manually. The nested struct is useful as future-proofing IMO, in case we decide to support rollback using some key other than revision, such as podTemplateHash. The controller should nil the whole `rollback` field.